src/pybind/mgr/nfs/export_utils.py

```diff
@@ -328,3 +382,57 @@ def to_dict(self):
             'fsal': self.fsal.to_dict(),
             'clients': [client.to_dict() for client in self.clients]
         }
+
+    @staticmethod
+    def validate_access_type(access_type):
+        valid_access_types = ['rw', 'ro', 'none']
+        if access_type.lower() not in valid_access_types:
+            raise NFSInvalidOperation(
+                f'{access_type} is invalid, valid access type are'
+                f'{valid_access_types}'
+            )
+
+    @staticmethod
+    def validate_squash(squash):
+        valid_squash_ls = [
+            "root", "root_squash", "rootsquash", "rootid", "root_id_squash",
+            "rootidsquash", "all", "all_squash", "allsquash", "all_anomnymous",
+            "allanonymous", "no_root_squash", "none", "noidsquash",
+        ]
+        if squash not in valid_squash_ls:
+            raise NFSInvalidOperation(
+                f"squash {squash} not in valid list {valid_squash_ls}"
+            )
+
+    def validate(self, mgr):
+        if not isabs(self.pseudo) or self.pseudo == "/":
+            raise NFSInvalidOperation(
+                f"pseudo path {self.pseudo} is invalid. It should be an absolute "
+                "path and it cannot be just '/'."
+            )
+
+        self.validate_squash(self.squash)
+        self.validate_access_type(self.access_type)
+
+        if not isinstance(self.security_label, bool):
+            raise NFSInvalidOperation('security_label must be a boolean value')
+
+        for p in self.protocols:
+            if p not in [3, 4]:
+                raise NFSInvalidOperation(f"Invalid protocol {p}")
+
+        valid_transport = ["UDP", "TCP"]
+        for trans in self.transports:
+            if trans.upper() not in valid_transport:
+                raise NFSInvalidOperation(f'{trans} is not a valid transport protocol')
+
+        for client in self.clients:
+            self.validate_squash(client.squash)
+            self.validate_access_type(client.access_type)
+
+        if self.fsal.name == 'CEPH':
+            fs = cast(CephFSFSAL, self.fsal)
+            if not check_fs(mgr, fs.fs_name):
+                raise FSNotFound(fs.fs_name)
+        else:
+            raise NFSInvalidOperation('FSAL {self.fsal.name} not supported')
```

Comment:
I would prefer this to be split up into different methods.  In export create command there is no need to validate everything.  For export update, we can have separate validate method which validates everything by calling up the split up methods. 

https://github.com/ceph/ceph/blob/cd0bb38b2b22d9f279a8b5d0b79243556b561f6f/src/pybind/mgr/nfs/export.py#L369-L399

Reply:
the single validate() method simplifies the code and makes the checks more robust--no chance of forgetting to validate a particular field piecemeal in the update